src/views/forms/FormMosaicDefinitionTransaction/FormMosaicDefinitionTransactionTs.ts

```diff
@@ -222,10 +223,14 @@ export class FormMosaicDefinitionTransactionTs extends FormTransactionBase {
      * Resetting the form when choosing a multisig signer and changing multisig signer
      * Is necessary to make the mosaic inputs reactive
      */
-    @Watch('selectedSigner')
-    onSelectedSignerChange() {
-        this.formItems.signerAddress = this.selectedSigner.address.plain();
-        if (this.isMultisigMode()) {
+    public async onChangeSigner(address: string) {
```

Comment:
Instead of overriding the behaviour in the base class, I think we should rename this method and call the inherited method instead. And also there is no need to call the namespaces API explicitly because it's already being done.https://github.com/symbol/desktop-wallet/blob/363e28a9f0baed42843f56c855e65296d451eb9c/src/store/Account.ts#L424
https://github.com/symbol/desktop-wallet/blob/363e28a9f0baed42843f56c855e65296d451eb9c/src/store/Namespace.ts#L156
```
public onFomSignerChange(address: string) {
    const signerChanged: boolean = this.formItems.signerAddress !== this.selectedSigner.address.plain();
    if (signerChanged) {
        this.onChangeSigner(address);
    }
    this.resetForm();
```
And are we 100% sure about the `signerChanged` boolean check, do we need it?
And most importantly, can you rebase your branch on top of `dev` ? I think some of the issues might be already solved.